### PR TITLE
Added viewmodel support to handle thread handling and screen stuck issue

### PR DESCRIPTION
**Pull request by rooparsh**
This PR adds the ViewModel which helps in fixing issue [Issue Link](https://github.com/ACM-VIT/gitpositive/issues/12).

### CODE REVIEW

**Comment by rooparsh** [comment]
@rishuyadav Please check and merge. If its ok, then will convert the other activity also

File: app/src/main/java/org/acmvit/gitpositive/MainActivity.kt [hunk 1]
```diff
@@ -1,80 +1,90 @@
...
 class MainActivity : AppCompatActivity() {
-    private var loadingDialog: AlertDialog? = null
+
+
+    private val viewModel by viewModels<MainViewModel>()
+    private var _binding: ActivityMainBinding? = null
+    private val binding: ActivityMainBinding
+        get() {
+            return _binding!!
+        }
```
Comment (DarthBenro008) [inline app/src/main/java/org/acmvit/gitpositive/MainActivity.kt:21]: Pretty smart to use a getter, but I would highly suggest to not use doublebang operator! There is a scope of optimisation here.

File: app/src/main/java/org/acmvit/gitpositive/MainViewModel.kt [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+import androidx.lifecycle.ViewModel
+import retrofit2.Call
+import retrofit2.Callback
+import retrofit2.Response
+import retrofit2.Retrofit
+import retrofit2.converter.gson.GsonConverterFactory
+
+/**
+ * Created by Rooparsh Kalia on 02/10/21
+ */
```
Comment (DarthBenro008) [inline app/src/main/java/org/acmvit/gitpositive/MainViewModel.kt:14]: Please remove authorship as this is an OpenSource code, for the community, by the community 😃

**Review by DarthBenro008 — CHANGES_REQUESTED** [review]
Well written code, left some comments. PTAL

File: app/src/main/java/org/acmvit/gitpositive/MainActivity.kt (hunk 1 above)
Comment (rooparsh) [reply app/src/main/java/org/acmvit/gitpositive/MainActivity.kt:21]: Hi @DarthBenro008 this one is backing field. 
Please find the reference here : [SO link](https://stackoverflow.com/questions/66442242/purpose-of-with-without-underscore-variable-view-binding)

**Comment by rooparsh** [comment]
Hi, @DarthBenro008, please check and let me know

**Review by DarthBenro008 — APPROVED** [review]
LGTM! Thank you for the PR!

Happy Hacktoberfest! 🎉

**Pull request merged**